frontend/src/viewer/components/2d/path-element.ts

```diff
@@ -319,6 +319,35 @@ export class PathElement extends connect(store)(LitElement) {
     return `http://xctrack.org/xcplanner?${params}`;
   }
 
+  private encodeNumber(num: number): string {
+    // google.maps.geometry.encoding.encodePath is for latLng only.
+    let pnum = num << 1;
+    if (num < 0) {
+      pnum = ~pnum;
+    }
+    let result = '';
+    while (pnum > 0x1f) {
+      result += String.fromCharCode(((pnum & 0x1f) | 0x20) + 63);
+      pnum = pnum >>> 5;
+    }
+    result += String.fromCharCode(63 + pnum);
+    return result;
+  }
+
+  private getXctsk(): string {
+    const path = this.line ? this.line.getPath() : [];
+    // https://xctrack.org/Competition_Interfaces.html#task-definition-format-2---for-qr-codes
+    const turnpoints = path.map((latLng: google.maps.LatLng, i: number) => ({
+      n: 'WPT ' + (i + 1),
+      z:
+        this.encodeNumber(1e5 * latLng.lng()) +
+        this.encodeNumber(1e5 * latLng.lat()) +
+        this.encodeNumber(0) + // Altitude
```

Comment:
It would be nice to pass `LatLonZ[]` instead and populate the altitude

Reply:
FlyXc couldn't specify height for a turnpoint, correct? Then I don't see an issue with passing 0 here which will be interpreted as ground.

Reply:
Altitude is added when the task is exported to a file.
The idea would be to re-use the same code.

Which make me think that this code should probably lives in common (code common to front and back where the files are generated)